artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -2727,7 +2731,7 @@ private Message makeCopy(final MessageReference ref,
       Message copy = message.copy(newID);
 
       if (copyOriginalHeaders) {
-         copy.referenceOriginalMessage(message, ref != null ? ref.getQueue().getName().toString() : null);
+         copy.referenceOriginalMessage(message, ref.getQueue().getName().toString());
```

Comment:
keep it ! :) nice one!

Reply:
@clebertsuconic thank you :) Can you please check again? Especially null reference logging.